src/mimefactory.rs

```diff
@@ -1000,17 +1000,6 @@ impl<'a> MimeFactory<'a> {
                             "Secure-Join".to_string(),
                             "vg-member-added".to_string(),
                         ));
-                        // FIXME: Old clients require Secure-Join-Fingerprint header. Remove this
-                        // eventually.
```

Comment:
This will break compatibility with versions of DC before https://github.com/deltachat/deltachat-core-rust/pull/3914, which was merged about a year ago?

In this case, it would be nice to wait two years before introducing this breaking change:

Some ways to distribute Desktop don't have any auto-update, so there may be some people using very old DCs. We start adding monthly device messages if Delta Chat is about a year outdated (see `DC_OUTDATED_WARNING_DAYS`). However, I don't know exactly when https://github.com/deltachat/deltachat-core-rust/pull/3914/ was actually released on all platforms, and unless we want to check it exactly, we should give some generous leeway for all users to notice this device message and update.

Reply:
Just in case -- the compatible version is 1.107.0.
> It is here in Desktop: https://github.com/deltachat/deltachat-desktop/commit/93644aa533f44e07b42499ef09eba492e1c1e40d
> and here in Android: https://github.com/deltachat/deltachat-android/commit/699d97cccc5450e83d4c67d9d58fce311e1458ea

So, it's ~11 months ago. But the problem is that the part of this compatibility code is already removed, e.g. the "*-received" Securejoin messages. So, the old clients don't see the Secure-Join-Fingerprint anymore in these messages and apparently don't mark the peer as verified on other devices (afair it's done only on the initial device when handling "*-request-with-auth"). So, afaiu the multi-device setup is already broken and removing this will also break the single-device one :) Also there is some compat code in `observe_securejoin_on_other_device()` removed in this PR which needs to be adjusted otherwise for the backward verification (but probably it's ok to just mark the peer as backward-verified optimistically).

Reply:
@hpk42 and @r10s both think it's fine to break compatibility (plus, it will take some time to publish the update), so I'll just consider this resolved